Add announcement shortcode and partial so the docs can display an announcement

aimeeu opened the pull request:
Add shortcode so we can display a link to the SIG Docs survey

I can get the notice to appear on the website home page but not on other pages. Yep, I really don't know what I'm doing... yet. Appreciate feedback!

/cc @zacharysarah

zacharysarah commented on config.toml:
```diff
@@ -78,6 +78,8 @@ nextUrl = "https://kubernetes-io-vnext-staging.netlify.com/"
 githubWebsiteRepo = "github.com/kubernetes/website"
 githubWebsiteRaw = "raw.githubusercontent.com/kubernetes/website"
 
+survey-notice = true
```
Try dropping the hyphen and adjusting _if_ statements accordingly. Hugo sometimes behaves oddly around punctuation in params.
> `surveynotice = true`

aimeeu replied:
That didn't make any difference. It's something with the way the docs/home page is constructed at compile time. I'm missing something fundamental. Is the person who put the Hugo site together still around?

jaypipes replied:
You need to drop the hyphen that then change the lines in the `layouts/shortcodes/survey-notice.html` and `layouts/partials/survey-notice.html` pages that look like this:

```html
{{ if .Param "survey-notice"}}
```

to this:

```html
{{ if .Param "surveynotice"}}
```

sftim commented on layouts/partials/survey-notice.html:
```diff
@@ -0,0 +1,11 @@
+    <main>
+      <div class="content survey-notice">
+        <h3>
+          The SIG Docs team needs your feedback! Please take a <a href="https://www.surveymonkey.com/r/8R237FN" target="_blank">short survey</a> so we can improve the Kubernetes documentation.
```
People reading this might click the link before finding that it's a survey about the docs rather than about feedback.
(“SIG Docs” should make that clear, but it might not).

If that's a concern, maybe:
```suggestion
          Tell us your views on this documentation. Please take a short <a href="https://www.surveymonkey.com/r/8R237FN" target="_blank">survey</a>&mdash;Kubernetes SIG Docs wants your feedback!
```

aimeeu replied:
Thanks for the suggestions!

sftim commented on static/css/survey-notice.css:
```diff
@@ -0,0 +1,6 @@
+    padding: 20px;
+    margin: 20px 0;
+    border-radius: 3px;
+    background-color: #eeeeee;
```
I recommend setting `color` as well. Maybe that's outdated advice but it was something I remember from CSS1.

aimeeu replied:
Kinda funny - I copied that code verbatim from deprecation-warning.css

sftim commented:
How about true camel case: `surveyNotice`?

jaypipes commented:
> How about true camel case: `surveyNotice`?

Sure, either way. Just needs to match what is put in the `.Params` value :)

aimeeu commented:
@jaypipes, @sftim  - I was always able to get the notice to appear on the kubernetes home page (/), just not the **docs** home page (/docs/home). Looks to me like the docs/home page must be generated at compile time. Since my knowledge of Hugo is basic (but growing), I haven't quite figured out what's going on. This isn't high priority, and as time is almost up for the survey, more of a "nice to know for next time" item.

aimeeu commented:
Made this generic: renamed survey-notice to announcement; added announcement_message to config.toml so in future, only announcement and announcement_message need to be changed

aimeeu commented:
Preview: https://deploy-preview-16210--kubernetes-io-master-staging.netlify.com/

zacharysarah commented:
This will be great for any future site announcements we need to make! ✨ 

/lgtm
/approve
